presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/MetastoreUtil.java

```diff
@@ -469,12 +469,38 @@ public static void renameFile(FileSystem fileSystem, Path source, Path target)
     // TODO: https://github.com/prestodb/presto/issues/15974
     public static Map<String, String> toPartitionNamesAndValues(String partitionName)
     {
-        try {
-            return ImmutableMap.copyOf(makeSpecFromName(partitionName));
+        ImmutableMap.Builder<String,String> resultBuilder = ImmutableMap.builder();
+        int index = 0;
+        int len = partitionName.length();
+        boolean invalidName = false;
+
+        while(index < len){
+            int keyStart = index;
+            while(index < len && partitionName.charAt(index) != '=') {
+                index++;
+            }
+            if(index >= len){
```

Comment:
Replace it with checkState(index < len, "Invalid partition spec: " + partitionName);